code/obj/artifacts/artifact_machines/food_processor.dm

```diff
@@ -0,0 +1,169 @@
+			var/list/obj/item/parts/convertable_limbs = keep_truthy(list(humanuser.limbs.l_arm, humanuser.limbs.r_arm, humanuser.limbs.l_leg, humanuser.limbs.r_leg))
+			//figure out which limbs are already robotic and remove them from the list
+			for (var/obj/item/parts/limb in convertable_limbs)
+				if (!limb || (limb.kind_of_limb & LIMB_ROBOT))
+					convertable_limbs -= limb
+			//people with existing robolimbs get converted faster.
+			//(loops_per_consumption_step - 1) bit adds some 'buffer time' before any limbs are converted.
+			var/loops = (loops_per_consumption_step * (convertable_limbs.len + 1)) + (loops_per_consumption_step - 1)
+			while (loops > 0)
+				if ((user.loc != O.loc && user.loc != O) || !activated)
+					src.food_processor_state = STATUS_READY
+					return
+				loops--
+				//inescapable version slices em up more
+				random_brute_damage(humanuser, (escapable ? 10 : 15))
+				take_bleeding_damage(humanuser, null, (escapable ? 3 : 4))
+				user.changeStatus("stunned", 7 SECONDS)
+				playsound(user.loc, pick(work_sounds), 50, 1, -1)
+				if (loops % loops_per_consumption_step == 0)
+					if (!convertable_limbs.len) //avoid runtiming once all limbs are converted
+						continue
+					var/obj/item/parts/limb_to_replace = pick(convertable_limbs)
+					switch(limb_to_replace.slot)
+						if ("l_arm")
+							qdel(humanuser.limbs.get_limb("l_arm"))
+						if ("r_arm")
+							qdel(humanuser.limbs.get_limb("r_arm"))
+						if ("l_leg")
+							qdel(humanuser.limbs.get_limb("l_leg"))
+						if ("r_leg")
+							qdel(humanuser.limbs.get_limb("r_leg"))
+					convertable_limbs -= limb_to_replace
+					humanuser.update_body()
+					src.biomatter += 2
+				sleep(0.4 SECONDS)
+
+			var/bdna = null // For forensics (Convair880).
+			var/btype = null
+			if (user.bioHolder.Uid && user.bioHolder.bloodType)
+				bdna = user.bioHolder.Uid
+				btype = user.bioHolder.bloodType
+			var/turf/T = get_turf(user)
+			gibs(T, null, bdna, btype)
+
+			ArtifactLogs(user, null, O, "touched", "food processing user", 0) // Added (Convair880).
+
+			//Biofuel per body copied from the enzymatic reclaimer
+			var/humanOccupant = (ishuman(user) && !ismonkey(user))
+			var/decomp = ishuman(user) ? user:decomp_stage : 0
+			src.biomatter += rand(5, 8) * (humanOccupant ? 2 : 1) * ((4.5 - decomp) / 4.5)
+
+			user.set_loc(get_turf(O.loc))
+			user.death()
+			user.ghostize()
+			qdel(user)
+
+			playsound(O.loc, 'sound/machines/pc_process.ogg', 50, 1, -1)
+			sleep(5 SECONDS)
+			src.spawn_food(O)
+			src.food_processor_state = STATUS_READY
+		else
+			return
+
+	proc/spawn_food(var/obj/O)
+		while(src.biomatter > src.biomatter_per_food)
+			playsound(O.loc, 'sound/machines/vending_dispense.ogg', 50, 1, -1)
+			sleep(0.5 SECONDS)
+
+			//Pick a random food item.
+			//TODO: Filter out raw ingredients as well as items requiring forks or other utensils.
+			src.biomatter -= src.biomatter_per_food
+			var/food = pick(src.possible_foods)
```

Comment:
Might be more interesting if it was the same food item each time per artifact? Or at least a chance for that?